executor/aggfuncs/func_count.go

```diff
@@ -329,7 +331,7 @@ func (e *countOriginalWithDistinct) evalAndEncode(
 		if err != nil || isNull {
 			break
 		}
-		encodedBytes = appendString(encodedBytes, buf, val)
+		encodedBytes = codec.EncodeBytes(encodedBytes, hack.Slice(val))
```

Comment:
Looks like `appendJSON` has risk as well?

Reply:
Yes, I'll change it.
But it a little difficult to write tests for it.

Reply:
I checked the [encode](https://github.com/pingcap/tidb/blob/master/util/codec/codec.go#L261-L265) and [decode](https://github.com/pingcap/tidb/blob/master/util/codec/codec.go#L403-L411) functions for JSON.
The encoded result for JSONs seems to be not comparable.
I create an issue https://github.com/pingcap/tidb/issues/10215 to record this, this would be fixed in an individual PR.